App/SB/views/FeedList/index.js

```diff
@@ -54,23 +57,56 @@ class Notifications extends React.PureComponent {
     }
   }
 
+  // gets called every time  the user enters this tab
+  _onFocus () {
+    // refresh the messages for the user
+    this.props.refreshNotifications() // < will get called on the very first entry too
+    // reset our readAll flag to re-time
+    this.setState({readAllOnExit: false})
+    // set a timer to clear all unread messages
+    this._setReadAllTimer()
+  }
+
+  // gets called every time the user exists the tab
+  _onBlur () {
+    // will be 0 if already run
+    if (this.readAllHandle) {
+      clearTimeout(this.readAllHandle)
+      this.readAllHandle = 0
+    }
+    // if the user was on the page long enough, we'll just clear all unread
+    if (this.state.readAllOnExit) {
+      this.props.readAllNotifications()
+    }
+  }
+
   componentDidMount () {
-    this.props.refreshNotifications()
+    // on mount, set listeners for enter and exit of the tab
+    this.props.navigation.addListener('willFocus', this._onFocus.bind(this))
+    this.props.navigation.addListener('willBlur', this._onBlur.bind(this))
```

Comment:
Ok. Based on what @sanderpick said, that isn't even necessary, right?

Reply:
We'll be refreshing messages when the node comes online, so this would be redundant I guess. He said to only call refresh messages on a pull to refresh kind of event.

Reply:
I _think_ when the user enters the Feed view we want to prompt an update for latest. but could mistaken.

Reply:
It's more of just what kind of interaction you'd want. I was just trying to reduce the number of calls that were made because they WILL get queued. That said, I just pushed this commit: https://github.com/textileio/textile-go/pull/266/commits/8adce43af6c319d7d644413569e5356e46b6631a which ensure they aren't. So, if that's in, it can be called however many times you want... subsequent calls will get ignored, not queued.